src/sql/workbench/contrib/notebook/browser/models/notebookContexts.ts

```diff
@@ -95,8 +95,9 @@ export class NotebookContexts {
 			if (connections && connections.length > 0) {
 				defaultConnection = connections[0];
 				if (profile && profile.options) {
-					if (find(connections, connection => connection.serverName === profile.serverName)) {
-						defaultConnection = find(connections, connection => connection.serverName === profile.serverName);
+					let matchingConn = find(connections, connection => connection.serverName === profile.serverName);
+					if (matchingConn) {
```

Comment:
Connections are unique by more than just their server name though - database and username also play a part. Is it expected behavior here that we're not checking anything else?

Reply:
I'm trying not to modify the original behavior too much. There were two duplicate find calls here, so I just combined them.

Reply:
That's fine - but it would be good to investigate whether this is an actual issue and if so open a bug for it. Seems like it might cause some problems when have multiple connections to the same server.

Reply:
Note that this isn't the one connection we return, it's just a default provided alongside the full list of connections.